c2rust-analyze/src/context.rs

```diff
@@ -665,6 +672,21 @@ impl<'tcx> GlobalAnalysisCtxt<'tcx> {
     pub fn iter_fns_failed(&self) -> impl Iterator<Item = DefId> + '_ {
         self.fns_failed.keys().copied()
     }
+
+    pub fn known_fn_ptr_perms<'a>(
+        &'a self,
+    ) -> impl Iterator<Item = (PointerId, PermissionSet)> + PhantomLifetime<'tcx> + 'a {
+        let tcx = &self.tcx;
+        self.fn_sigs
```

Comment:
I might not be remembering this correctly, but don't we already keep track of the foreign signatures? Do you need to filter them here?

Reply:
Not in a separate map, I don't think. `gather_foreign_sigs` adds its `LFnSig`s to `gacx.fn_sigs`.

Reply:
Ah right, I had thought about making a separate map but folded it into `fn_sigs` after all